client/extensions/woocommerce/woocommerce-services/state/shipping-label/actions.js

```diff
@@ -850,3 +841,17 @@ export const confirmReprint = ( orderId, siteId ) => ( dispatch, getState ) => {
 		} )
 		.then( () => dispatch( closeReprintDialog( orderId, siteId ) ) );
 };
+
+export const updatePaperSize = ( orderId, siteId, value ) => ( dispatch, getState ) => {
+	dispatch( {
+		type: WOOCOMMERCE_SERVICES_SHIPPING_LABEL_UPDATE_PAPER_SIZE,
+		siteId,
+		orderId,
+		value,
+	} );
+
+	const shippingLabel = getShippingLabel( getState(), orderId, siteId );
+	if ( shippingLabel.reprintDialog != null ) {
+		dispatch( openReprintDialog( orderId, siteId, shippingLabel.reprintDialog.labelId ) );
```

Comment:
This will be unstable because changing the paper size will trigger re-downloading the PDF, but won't cancel the previous download.

What do you think about checking [here](https://github.com/Automattic/wp-calypso/pull/18930/files#diff-09d52b7e14fcd8c2a349cabb5e073666R825) that the `paperSize` is still the same, and if not, don't do anything?

Reply:
Done — let me know if something clearer than `shippingLabelAfter` comes to mind.